pageserver/src/tenant/storage_layer/layer.rs

```diff
@@ -860,6 +860,21 @@ impl LayerInner {
                         Ok(())
                     }
                     Err(e) => {
+                        let consecutive_failures =
+                            this.consecutive_failures.fetch_add(1, Ordering::Relaxed);
+
+                        let backoff = utils::backoff::exponential_backoff_duration_seconds(
+                            consecutive_failures.min(u32::MAX as usize) as u32,
+                            1.5,
+                            60.0,
+                        );
+                        let backoff = std::time::Duration::from_secs_f64(backoff);
+
+                        tokio::select! {
+                            _ = tokio::time::sleep(backoff) => {},
```

Comment:
Can this use tokio's Timeout functionality?

Reply:
It could now, but I realized we need to cancel by the timeline cancellation token as well (similarly to #6133) in https://github.com/neondatabase/neon/pull/5746#discussion_r1428159294 since but of course #6134 has other TODO as well.